anvil/src/eth/api.rs

```diff
@@ -490,6 +493,25 @@ impl EthApi {
         self.backend.block_by_number_full(number).await
     }
 
+    /// Returns blocks with given numbers array.
+    ///
+    /// Handler for ETH RPC call: `eth_getBlocksByArray`
+    pub async fn blocks_by_array(
+        &self,
+        numbers: Vec<BlockNumber>,
+    ) -> Result<Option<Vec<Block<Transaction>>>> {
+        node_info!("eth_getBlocksByArray");
+
+        let mut blocks: Vec<Block<Transaction>> = Vec::new();
+        for number in numbers {
+            if let Some(block) = self.block_by_number_full(number).await? {
+                blocks.push(block);
+            }
```

Comment:
this is only async because we need to handle cases where the number predates the block we forked off from.

but we can put this `join_all` here, nonetheless.